### PR TITLE
Refactor commodity repository and update database schema

### LINKED ISSUE #157: Merge `commodity` and `commodity_whitelist` Tables to Improve Query Performance
### Description
In our current database schema, we have two separate tables: commodity and commodity_whitelist. These tables are interconnected and used to create an intermediary view named validated_commodity_view to facilitate our queries. This setup, while functional, has been identified as a bottleneck in our system, contributing to slower query performances due to the overhead associated with maintaining and querying through the intermediary view.

### Objective
The primary goal of this ticket is to merge the commodity and commodity_whitelist tables into a single, efficiently structured table. This consolidation aims to eliminate the need for the validated_commodity_view and, as a result, speed up our query processes by reducing complexity and improving data retrieval times.

### Expected Benefits
- Performance Improvement: Direct queries against a single table will be faster than querying through an intermediary view, especially as our data volume continues to grow.
- Simplified Schema: Having one less view and one less table in our database schema simplifies understanding and maintaining our database.
- Scalability: A simplified and more efficient schema lays a better foundation for scaling our application as needed.

### PR DESCRIPTION
**Pull request by pveeckhout**
Updated MybatisCommodityRepository, replacing several methods with a createOrUpdateOnConflict method and added new pertinent fields to the database table "commodity". This includes adding and renaming columns, updating the "validated_commodity_view" and adding an index for the "is_rare" column. In "build.gradle", the backendMessageProcessorLibVersion has been updated.

closes #157

### CODE REVIEW

File: trade-module/src/main/resources/db/trademodule/changelog/create_validated_commodity_view.xml
Comment (Daniel-J-Mason) [inline trade-module/src/main/resources/db/trademodule/changelog/create_validated_commodity_view.xml:1]: This breaks liquibase changelog checksum, have to recreate in a new file. maybe something like update_validated_commodity_view.xml

May come up with a standard naming system if we have multiple updates

**Review by Daniel-J-Mason — COMMENTED** [review]
validated_commodity may be useful to query, but in the spirit of merging these entities I think the entire ValidatedCommodity application class should be removed from the backend stack. The Commodity class should be updated to include the full field set and be the sole reference so cut down on confusion. We havent separated any of our other classes based on completeness either so this would be a more consistent approach

File: trade-module/src/main/resources/db/trademodule/changelog/create_validated_commodity_view.xml
Comment (pveeckhout) [reply trade-module/src/main/resources/db/trademodule/changelog/create_validated_commodity_view.xml:1]: this should not break the checksum as it should always do a recreate upon start.

**Review by Daniel-J-Mason — APPROVED** [review]
View corrected

maintaining ValidatedCommodity architecture for the time being as its the only domain object we have intentionally null data to distinguish validity

**Pull request merged**